wikilink/wiki_link.py

```diff
@@ -525,3 +570,26 @@ def _insert_link(session, from_page_id, to_page_id, no_of_separation):
 			"Many rows found in DB to store link from {} to {}\
 			 with number of seperation {}".format(from_page_id, to_page_id,
 		 								          no_of_separation))
+
+class LinkNode(object):
+    def __init__(self, node_id):
+        self.node_id = node_id
+        self.path = [] #parent urls
+
+def get_all_my_parents(node_id, session):
+    parent_list = []
+    for every_parent in session.query(Link).filter(Link.to_page_id==node_id,Link.number_of_separation==1):
+        parent_node = LinkNode(every_parent.from_page_id)

```

Comment:
For DB operation, please consider to add try/except, i.e. in case DB connection failure